### DISCUSSION TITLE
Don't accept trades from these banks (SEPA accounts)

### DISCUSSION THREAD

**Discussion by Android-X13** [bisq-network/bisq#6290] [Ideas]
I think it would be a good feature to have in the Account settings.

For example if one wanted to avoid trading with peers that belong to a specific bank, because it is known that the bank is making trouble (freezing transfers & asking for sender/receiver personal details to continue, etc.). As it is now you can only reject countries, so if you really want to reject a Norwegian bank (as an example) you'll need to reject Norway as a whole to avoid traders from that one bank (unless I'm terribly mistaken).

Since they are too many it would probably be impractical to maintain an internal database of all known banks as well as a very long list of check boxes or a huge dropdown menu containing them in the GUI. So the option could simply be "Reject trades from these banks" and the user could specify one or more BICs to be rejected, depending the number of banks he doesn't want to accept trades from.
I think it's trivial for anyone to find a bank's BIC online.

**Comment by pazza83** [comment]
Sounds good to me. 

Might be a big job though?

BICs are mainly for EUR, others bank identifiers might also need to be considered.

Not sure if the number of users that would make use of this feature would justify the cost of implementing it?

**Reply by Android-X13** [reply]
In my case with one offer of mine in Bisq, I had to reject a whole country to avoid a bank. Ironically, when my offer was taking too long to be taken and decided to re-accept the country, the offer was taken from that country... fortunately from another bank :grin:

> BICs are mainly for EUR, others bank identifiers might also need to be considered.

I don't know much about banks unfortunately. I thought they use BICs internationally but sometimes they're called SWIFT codes (but they're the same thing)? 
Though I was refering only to the SEPA or SEPA Instant payment methods. In Bisq these two only have a field for BIC, so they are eligible for such an option.

I'm trying to familiarize with Bisq's code these days and I think it wouldn't be much of a hassle to implement this. And I'm not at the level of other experienced coders here..
